### PR TITLE
Add ValidationAttributeError support

### PR DESCRIPTION
**Pull request by km256**

### CODE REVIEW

File: lib/model.js [hunk 1]
```diff
@@ -198,15 +198,19 @@ var EventEmitter = require('eventemitter3'),
...
                 } else {
-                    return Vow.reject(validateResults.reduce(function (err, isValid, k) {
-                        if (!isValid) {
-                            err.attributes.push(model._attributesAr[k]);
+                    return Vow.reject(validateResults.reduce(function (err, validationResult, k) {
+                        if (validationResult !== true) {
+                            if (validationResult === false) {
+                                validationResult = new model.__self.ValidationAttributeError(model._attributesAr[k]);
+                            }
+                            err.attributes.push(validationResult);
```
Comment (mdidkivskyi) [inline lib/model.js]: What if validation method returns string?

Comment (km256) [inline lib/model.js]: Model validation promise wil be rejectected with array of strings

File: lib/model.js [hunk 2]
```diff
@@ -198,15 +198,19 @@ var EventEmitter = require('eventemitter3'),
                     return attribute.validate();
                 }));
             }).then(function (validateResults) {
-                if (
-                    //return 0 if any is false
-                    Math.min.apply(Math, validateResults)
-                ) {
+                var isValid = validateResults.every(function(result) {
```
Comment (anton-kotenko) [inline lib/model.js]: validateResults.every(Boolean)

**Comment by anton-kotenko** [comment]
Looks good to me, except small non important fix.
Also it would be great if you update a little documentation with description of validate method.

File: lib/model.js (hunk 2 above)
Comment (km256) [inline lib/model.js]: it's wrong, because `Boolean(new Error()) === true`

**Pull request merged**